dqn_main.cpp

```diff
@@ -73,51 +73,59 @@ Action GetAction(int action_idx) {
  */
 double PlayOneEpisode(HFOEnvironment& hfo, dqn::DQN& dqn, const double epsilon,
                       const bool update) {
-  // CHECK(!ale.game_over());
-  // std::deque<dqn::FrameDataSp> past_frames;
-  // auto total_score = 0.0;
-  // for (auto frame = 0; !ale.game_over(); ++frame) {
-  //   const ALEScreen& screen = ale.getScreen();
-  //   const auto current_frame = dqn::PreprocessScreen(screen);
-  //   past_frames.push_back(current_frame);
-  //   if (past_frames.size() < dqn::kInputFrameCount) {
-  //     // If there are not past frames enough for DQN input, just select NOOP
-  //     for (auto i = 0; i < FLAGS_skip_frame + 1 && !ale.game_over(); ++i) {
-  //       total_score += ale.act(PLAYER_A_NOOP);
-  //     }
-  //   } else {
-  //     while (past_frames.size() > dqn::kInputFrameCount) {
-  //       past_frames.pop_front();
-  //     }
-  //     dqn::InputFrames input_frames;
-  //     std::copy(past_frames.begin(), past_frames.end(), input_frames.begin());
-  //     const auto action = dqn.SelectAction(input_frames, epsilon);
-  //     auto immediate_score = 0.0;
-  //     for (auto i = 0; i < FLAGS_skip_frame + 1 && !ale.game_over(); ++i) {
-  //       immediate_score += ale.act(action);
-  //     }
-  //     total_score += immediate_score;
-  //     // Rewards for DQN are normalized as follows:
-  //     // 1 for any positive score, -1 for any negative score, otherwise 0
-  //     const auto reward = immediate_score == 0 ? 0 : immediate_score /
-  //         std::abs(immediate_score);
-  //     assert(reward <= 1 && reward >= -1);
-  //     if (update) {
-  //       // Add the current transition to replay memory
-  //       const auto transition = ale.game_over() ?
-  //           dqn::Transition(input_frames, action, reward, boost::none) :
-  //           dqn::Transition(input_frames, action, reward,
-  //                           dqn::PreprocessScreen(ale.getScreen()));
-  //       dqn.AddTransition(transition);
-  //       // If the size of replay memory is large enough, update DQN
-  //       if (dqn.memory_size() > FLAGS_memory_threshold) {
-  //         dqn.Update();
-  //       }
-  //     }
-  //   }
-  // }
-  // ale.reset_game();
-  // return total_score;
+  std::deque<dqn::StateDataSp> past_states;
+  hfo_status_t status = IN_GAME;
+  while (status == IN_GAME) {
+    const auto current_state = hfo.getState();
+    dqn::StateData current_state_;
+    std::copy(current_state.begin(), current_state.begin() + dqn::kStateDataSize,
```

Comment:
We need to check that the length of the state vector actual equals dqn::kStateDataSize. This can be accomplished using CHECK_EQ(current_state.size(), dqn::kStateDataSize);